stream-parent/stream-core/src/main/java/org/interledger/stream/AmountTooLargeErrorData.java

```diff
@@ -0,0 +1,35 @@
+package org.interledger.stream;
+
+import org.interledger.core.Immutable;
+
+import com.google.common.primitives.UnsignedLong;
+
+/**
+ * An error returned by a Connector when a packet amount is too large.  Each of the following should be included as the
+ * `data` field in an ILP Reject packet.
+ */
+public interface AmountTooLargeErrorData {
+
+  static AmountTooLargeErrorDataBuilder builder() {
+    return new AmountTooLargeErrorDataBuilder();
+  }
+
+  /**
+   * Local amount received by the connector.
+   *
+   * @return A {@link UnsignedLong} with the received amount.
+   */
+  UnsignedLong receivedAmount();
+
+  /**
+   * Maximum amount (inclusive, denominated in same units as the receivedAmount) the connector will forward.
+   *
+   * @return A {@link UnsignedLong} with the maximum amount.
+   */
+  UnsignedLong maximumAmount();
+
+  @Immutable
```

Comment:
Any reason to create an abstract class here instead of just annotating the interface as `@Immutable`?

Reply:
> Any reason to create an abstract class here

Not really - the original idea was that the interfaces could be broken out to their own library one day and maybe not have to rely upon immutables (with the abstract classes living in some other library). 

However, since immutables is really just a thing engaged during compile time (`provided` scope) I think that ship has sailed. So, unless something is required in an abstract class, using one isn't really adding any value.